File: biblionode-api/src/main/java/io/github/mgrablo/BiblioNode/dto/BookResponse.java
package io.github.mgrablo.BiblioNode.dto;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.time.LocalDateTime;

public record BookResponse(
		Long id,
		String title,
		String isbn,
		String authorName,
		Long authorId,
		boolean available,

		@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
		LocalDateTime createdAt,

		@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
		LocalDateTime modifiedAt
) { }

File: biblionode-ui/src/app/features/books/data-access/dto/book-response.ts
export interface BookResponse {
  id: number;
  title: string;
  isbn: string;
  authorName: string;
  authorId: number;
  available: boolean;
  coverUrl: string | null;
  description: string | null;
  createdAt: string;
  modifiedAt: string;
}
